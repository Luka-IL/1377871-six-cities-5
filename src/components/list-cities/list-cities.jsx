import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/action";


const ListCities = (props) => {

  const {changeCity} = props;
  const cities = [`Amsterdam`, `Paris`, `Cologne`, `Brussels`, `Hamburg`, `Dusseldorf`];


  const cityClickHandler = (evt) => {
    evt.preventDefault();
    changeCity(evt.target.id);
  };

  return (
    <ul className="locations__list tabs__list">
      {cities.map((item, i) => (
        <li key={`city-${i}`} className="locations__item">
          <a className="locations__item-link tabs__item" href="#" id={item} onClick={cityClickHandler}>
            {item}
          </a>
        </li>
      ))}
    </ul>
  );
};

ListCities.propTypes = {
  changeCity: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  changeCity(city) {
    dispatch(ActionCreator.changeCity(city));
  },
});

export {ListCities};
export default connect(null, mapDispatchToProps)(ListCities);
