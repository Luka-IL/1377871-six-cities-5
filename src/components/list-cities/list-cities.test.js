import React from "react";
import renderer from "react-test-renderer";
import {ListCities} from "./list-cities";

it(`Should Favorites render correctly`, () => {
  const tree = renderer
    .create(<ListCities
      changeCity={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
