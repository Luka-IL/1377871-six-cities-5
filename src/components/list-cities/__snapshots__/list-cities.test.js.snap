// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Favorites render correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      id="Amsterdam"
      onClick={[Function]}
    >
      Amsterdam
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      id="Paris"
      onClick={[Function]}
    >
      Paris
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      id="Cologne"
      onClick={[Function]}
    >
      Cologne
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      id="Brussels"
      onClick={[Function]}
    >
      Brussels
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      id="Hamburg"
      onClick={[Function]}
    >
      Hamburg
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      id="Dusseldorf"
      onClick={[Function]}
    >
      Dusseldorf
    </a>
  </li>
</ul>
`;
